include/seqan3/range/container/bitcompressed_vector.hpp

```diff
@@ -0,0 +1,1101 @@
+
+    /*!\brief Returns the maximum number of elements the container is able to hold due to system or library
+     * implementation limitations, i.e. std::distance(begin(), end()) for the largest container.
+     * \returns The number of elements in the container.
+     *
+     * This value typically reflects the theoretical limit on the size of the container. At runtime, the size
+     * of the container may be limited to a value smaller than max_size() by the amount of RAM available.
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    size_type max_size() const noexcept
+    {
+        return data.max_size();
+    }
+
+    /*!\brief Returns the number of elements that the container has currently allocated space for.
+     * \returns The capacity of the currently allocated storage.
+     *
+     * \attention
+     *
+     * This does not operate on underlying concat container, see concat_capacity().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    size_type capacity() const noexcept
+    {
+        return data.capacity();
+    }
+
+    /*!\brief Increase the capacity to a value that's greater or equal to new_cap.
+     * \param            new_cap The new capacity.
+     * \throws std::length_error If new_cap > max_size().
+     * \throws    std::exception Any exception thrown by `Allocator::allocate()` (typically `std::bad_alloc`).
+     *
+     * Increase the capacity of the vector to a value that's greater or equal to new_cap.
+     * If new_cap is greater than the current capacity(), new storage is allocated, otherwise the method does nothing.
+     * If new_cap is greater than capacity(), all iterators, including the past-the-end iterator, and all references
+     * to the elements are invalidated. Otherwise, no iterators or references are invalidated.
+     *
+     * ### Complexity
+     *
+     * At most linear in the size() of the container.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    void reserve(size_type const new_cap)
+    {
+        data.reserve(new_cap);
+    }
+
+    /*!\brief Requests the removal of unused capacity.
+     *
+     * It is a non-binding request to reduce capacity() to size() and concat_capacity() to concat_size().
+     * It depends on the implementation if the request is fulfilled.
+     * If reallocation occurs, all iterators, including the past the end iterator, and all references to the elements
+     * are invalidated. If no reallocation takes place, no iterators or references are invalidated.
+     *
+     * ### Complexity
+     *
+     * At most linear in the size() of the container.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    void shrink_to_fit()
```

Comment:
can't we assume noexcept?
First, destruction cannot cause an exception. Second, memory is only freed and `delete` is actually documented as `noexcept` since c++11.

Reply:
I wasn't sure whether shrink_to_fit might not induce reallocation. Also I don't so a potential for optimisation so I would rather not add `noexcept` there now.